tests/Silverback.Integration.Tests.E2E/Kafka/MessageValidationTests.cs

```diff
@@ -0,0 +1,344 @@
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedKafka())
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint
+                                        .WithValidationMode(ValidationMode.None)
+                                        .ProduceTo(DefaultTopicName)))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+
+            // Act
+            Func<Task> act = () => publisher.PublishAsync(message);
+
+            // Assert
+            await act.Should().NotThrowAsync<ValidationException>();
+            DefaultTopic.MessagesCount.Should().Be(1);
+        }
+
+        [Fact]
+        public async Task Validation_WithWarning_InvalidMessageIsProduced()
+        {
+            // Arrange
+            var message = new Message
+            {
+                String10 = "1234567890abcd"
+            };
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedKafka())
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint
+                                        .WithValidationMode(ValidationMode.WithWarning)
+                                        .ProduceTo(DefaultTopicName)))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+
+            // Act
+            Func<Task> act = () => publisher.PublishAsync(message);
+
+            // Assert
+            await act.Should().NotThrowAsync<ValidationException>();
+            DefaultTopic.MessagesCount.Should().Be(1);
+        }
+
+        [Fact]
+        public async Task Validation_WithException_InvalidMessageNotConsumed()
```

Comment:
We should also test that the exception bubbles up and the consumer gets stopped.